src/porepy/geometry/point_in_polyhedron_test.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+
+
+class PointInPolyhedronTest(object):
+    """This class implements a robust point in polyhedron test supporting non-convex
+        polyhedra.
+
+    The implementation is based on the reference: Robust inside-outside segmentation
+        using generalized winding numbers (https://doi.org/10.1145/2461912.2461916)
+
+    Implementation requires consistent orientation of the triangulated surface.
+
+    Attributes:
+        vertices (np.ndarray, num_pt x 3): Triangulation vertices.
+        connectivity (np.ndarray num_triangles x 3): Triangulation connectivity map.
+        tol (float, optional): Geometric tolerance, used in comparison of
+            points, areas and volumes. Defaults to 1e-10.
+    """
+
+    def __init__(
+        self,
+        vertices: np.ndarray[Any, np.dtype[np.float64]],
+        connectivity: np.ndarray[Any, np.dtype[np.int64]],
+        tol: float = 1e-10,
+    ):
+        self.vertices: np.ndarray[Any, np.dtype[np.float64]] = vertices
+        self.connectivity: np.ndarray[Any, np.dtype[np.int64]] = connectivity
+        self.tol: float = tol
+
+    def solid_angle(self, R: np.ndarray[Any, np.dtype[np.float64]]) -> float:
+        """Implements an analytical expression for the solid angle.
+
+        The solid angle is described by a plane triangle surface S at some arbitrary
+        point P. It is the measure of the intersection of the three-dimensional unit
+        sphere and the conic hull defined by the vectors in R.
+
+        The implementation is based on equation (6) in reference: Robust inside-outside
+            segmentation using generalized winding numbers
+            (https://doi.org/10.1145/2461912.2461916)
+
+        Args:
+            R (np.ndarray, num_pt x 3): Triangle points relocated to the origin point
+                [0,0,0]. Since the unit sphere is centered at P, we translated the
```

Comment:
The past tense is a bit confusing. Does this mean that the method should only be called for translated triangles?

Reply:
Agree. I have rewritten those lines.